### PR TITLE
Update docs about adding template

### PR DESCRIPTION
**Pull request by vicbergquist**
**What kind of change does this PR introduce?**
Added more info about creating a template and the steps involved.

**What is the current behavior?**
Noticed when I created a template that clearer docs would be nice.

**What is the new behavior?**
More info about creating a template.

**Checklist**:

- [x] Documentation
- [x] Tests
- [ ] Ready to be merged 
- [x] Added myself to contributors table 

I might have missed something or not explained something properly/in the wrong way, so would love some feedback.

### CODE REVIEW

**Comment by SaraVieira** [comment]
Bless your heart!

Bogdan, any stack trace for this fail?

File: contributor-docs/adding-template.md [hunk 1]
```diff
@@ -1,85 +1,180 @@
-# Adding a Template
+# Contribute with a Template
 
-This doc will describe all the resource you might need to add a new template to CodeSandbox. We will list all the contributions that are needed to add a new template, and will give some examples.
+To contribute with a CodSandbox template, you need to complete multiple steps and submit two Pull Requests and we have created this guide to help you do thisTo do this. Here you will find descriptions and explainations for what you need to do, along with some examples that we have added as references.
```
Comment (SaraVieira) [inline contributor-docs/adding-template.md]:
```suggestion
To contribute with a CodSandbox template, you need to complete multiple steps and submit two Pull Requests and we have created this guide to help you do this. Here you will find descriptions and explanations for what you need to do, along with some examples that we have added as references.
```

**Comment by SaraVieira** [comment]
Bless your heart!

@lbogdan any stack trace for this fail?

File: contributor-docs/adding-template.md [hunk 2]
```diff
@@ -1,85 +1,180 @@
...
+Unlike sandboxes, **containers** are executed on a server. This makes it possible to create projects that are end-to-end based, like `Next.js` with CodeSandbox, and also makes it possible to build bigger projects. **Containers** let you run any command, and everything that works locally will also work in container.
 
-There are some limitations to this approach though: we only allow signed in users to work on a container and the amount of container sandboxes a user can have is limited (this limit changes as we make containers more stable). You also can't edit containers while offline and it's not possible to edit containers when viewing from an embed.
+However, like the sandboxes, **containers** also come with some limitations. In order to work on a container, you need to be signed in as a user, you can't edit containers while offline, it's not possible viewing them from an embed, and ~~you can only have a limited amount of container based projects~~ ([containers are now out of beta](https://medium.com/@compuives/codesandbox-containers-out-of-beta-improvements-be79679df4cc)).
```
Comment (SaraVieira) [inline contributor-docs/adding-template.md]: Containers are out of beta but we still have a limited amount a user can have :/

File: contributor-docs/adding-template.md [hunk 3]
```diff
@@ -1,85 +1,180 @@
...
 
 ### Which one to choose?
 
-The template defines whether a sandbox is executed in a container or in the browser. This means that it's your choice where to execute the sandbox.
+As you have read above, the template type determines whether a project is executed in a *sandbox in a browser* or in a *container on a server*. This means that the template type you have to choose, depends on your specific use-case and where you want your project to be executed.
+
+If you would like to demonstrate CLI functionality, we recommend that you use a **container** type template, and if you want to demonstrate a JavaScript framework (like `create-react-app`) we recommended using a**sandbox** type template.
```
Comment (SaraVieira) [inline contributor-docs/adding-template.md]:
```suggestion
If you would like to demonstrate CLI functionality, we recommend that you use a **container** type template, and if you want to demonstrate a JavaScript framework (like `react`) we recommended using a**sandbox** type template.
```

Create React App is the CLI, maybe makes sense for this to be react

File: contributor-docs/adding-template.md [hunk 4]
```diff
@@ -1,85 +1,180 @@
-# Adding a Template
+# Contribute with a Template
 
-This doc will describe all the resource you might need to add a new template to CodeSandbox. We will list all the contributions that are needed to add a new template, and will give some examples.
+To contribute with a CodSandbox template, you need to complete multiple steps and submit two Pull Requests and we have created this guide to help you do this. Here you will find descriptions and explanations for what you need to do, along with some examples that we have added as references.
```
Comment (lbogdan) [inline contributor-docs/adding-template.md]:
```suggestion
To contribute with a CodeSandbox template, you need to complete multiple steps and submit two Pull Requests and we have created this guide to help you do this. Here you will find descriptions and explanations for what you need to do, along with some examples that we have added as references.
```

File: contributor-docs/adding-template.md [hunk 5]
```diff
@@ -1,85 +1,180 @@
...
+We realise that the process of submitting a template is not straightforward and we hope that this guide will help you along the way. We are working on a system that will make it easier.
 
-## What are templates?
+If you think that we missed something in this guide or believe we could explain something better, please let us know by submitting an [issue](https://github.com/CompuIves/codesandbox-client/issues/new/choose) with your feedback.
 
-Templates are nothing more than an identifier for a sandbox. We use this identifier to improve the UX in the editor and the preview. For example, for the `vue-cli` template we change the default `eslint` rules and what can be configured in a sandbox. We encourage every template creator to take an effort to improve the editor experience for their template, that will give the best experience for everyone using your template.
+## What is a template?
 
-## Types of Templates
+A template is an identifier for a specific type of sandbox project that you can create on [codesandbox.io](https://codesandbox.io), like projects using `Gatsby`, `React` or `Vue`.
```
Comment (lbogdan) [inline contributor-docs/adding-template.md]:
```suggestion
A template is an identifier for a specific type of sandbox project that you can create on [codesandbox.io](https://codesandbox.io), like projects using `Gatsby`, `React` or `Vue.js`.
```

File: contributor-docs/adding-template.md (hunk 3 above)
Comment (lbogdan) [reply contributor-docs/adding-template.md]: `React`, capital "r". 🙂

File: contributor-docs/adding-template.md (hunk 2 above)
Comment (lbogdan) [reply contributor-docs/adding-template.md]: What @SaraVieira said, we should drop the strike-through to not mislead people. As a side-note, client sandboxes are also limited for regular users, only the limit is higher (50 vs. 15).

File: contributor-docs/adding-template.md [hunk 6]
```diff
@@ -1,85 +1,180 @@
...
+
+If you would like to demonstrate CLI functionality, we recommend that you use a **container** type template, and if you want to demonstrate a JavaScript framework (like `create-react-app`) we recommended using a**sandbox** type template.
+
+We encourage everyone to first evaluate whether the template works as a sandbox, before deciding on using a container.
+
+## Adding a new template
+
+In order to add a new template, you need to go through a set of steps. Some of these steps you have to do, others depend on the type of template you want to add (**sandbox** vs. **container**).
+
+To get started, you should first follow the steps in our Contribution guidelines in order to [set up CodeSandbox locally](https://github.com/CompuIves/codesandbox-client/blob/master/CONTRIBUTING.md#setting-up-the-project-locally).
```
Comment (lbogdan) [inline contributor-docs/adding-template.md]:
```suggestion
To get started, you should first follow the steps in our contribution guidelines in order to [set up CodeSandbox locally](https://github.com/CompuIves/codesandbox-client/blob/master/CONTRIBUTING.md#setting-up-the-project-locally).
```

File: contributor-docs/adding-template.md [hunk 7]
```diff
@@ -1,85 +1,180 @@
...
+Example:
+
+```js
+import * as React from 'react';
+import filename from './filename.png';
+
+export default props => <img alt="filename" src={filename} {...props} />;
+```
+
+Replace `filename` with your logo name. The file name should be all lowercase and the `.tsx` file should be capitalised: `Filename.tsx`.
```
Comment (lbogdan) [inline contributor-docs/adding-template.md]:
```suggestion
Replace `filename` with your logo name. The image file name should be all lowercase and the `.tsx` file should be capitalised: `Filename.tsx`. 
```

**Comment by lbogdan** [comment]
Hey @vicbergquist ,

Great stuff right here, thanks! I left a few tiny suggestions.

Also, I'm a bit on the fence regarding terminology - sandbox / client sandbox / browser sandbox / container sandbox / server sandbox, which has been going on since we launched containers, so this might be a good opportunity to settle this once and for all! 🙂 
Cc @CompuIves ^^

**Comment by vicbergquist** [comment]
> Hey @vicbergquist ,
> 
> Great stuff right here, thanks for your effort! I left a few tiny suggestions.
> 
> Also, I'm a bit on the fence regarding terminology - sandbox / client sandbox / browser sandbox / container sandbox / server sandbox - which has been going on since we launched containers, so this might be a good opportunity to settle this once and for all! 🙂
> Cc @CompuIves ^^

thank you!

Yes, it is so confusing :/

File: contributor-docs/adding-template.md (hunk 2 above)
Comment (vicbergquist) [reply contributor-docs/adding-template.md]: Fixed 😊

File: contributor-docs/adding-template.md [hunk 8]
```diff
@@ -1,85 +1,180 @@
...
+CodeSandbox executes projects in the browser, which we call sandboxes. This means that the transpiling, bundling, dependency resolvment and more happens in the browser itself, without a server being involved. This has some advantages over conventional approaches; it works offline, is more performant and doesn't give us server costs, which means we can have many sandboxes without having to worry (a lot ;-)).
 
-There are also some disadvantages to running the sandboxes in a browser. The main one is that we lose flexibility: it's not possible to run CLI commands and in some cases we don't support custom configuration (like a custom webpack configuration). That's why we developed a new kind of sandbox called Container in September.
+There are also some disadvantages to this approach. When sandboxes run in the browser, we lose flexibility. It's no longer possible to run Command Line Interface (CLI) commans and in some cases custom configations are not supported, for example a custom webpack configuration. This is why we have developed a new kind of sandbox called **Container**, which we released in September 2018.
```
Comment (CompuIves) [inline contributor-docs/adding-template.md]:
```suggestion
There are also some disadvantages to this approach. When sandboxes run in the browser, we lose flexibility. It's no longer possible to run Command Line Interface (CLI) commands and in some cases custom configurations are not supported, for example a custom webpack configuration. This is why we have developed a new kind of sandbox called **Container**, which we released in September 2018.
```

File: contributor-docs/adding-template.md [hunk 9]
```diff
@@ -1,85 +1,180 @@
...
+Unlike sandboxes, **containers** are executed on a server. This makes it possible to create projects that are end-to-end based, like `Next.js` with CodeSandbox, and also makes it possible to build bigger projects. **Containers** let you run any command, and everything that works locally will also work in container.
 
-There are some limitations to this approach though: we only allow signed in users to work on a container and the amount of container sandboxes a user can have is limited (this limit changes as we make containers more stable). You also can't edit containers while offline and it's not possible to edit containers when viewing from an embed.
+However, like the sandboxes, **containers** also come with some limitations. In order to work on a container, you need to be signed in as a user, you can't edit containers while offline, it's not possible viewing them from an embed, and you can only have a limited amount of container based projects.
```
Comment (CompuIves) [inline contributor-docs/adding-template.md]:
```suggestion
However, like the sandboxes, **containers** also come with some limitations. In order to work on a container, you need to be signed in as a user, you can't edit containers while offline, it's not possible to edit them from an embed, and you can only have a limited amount of container based projects.
```

**Comment by vicbergquist** [comment]
> Wow, this is really good @vicbergquist. Thank you so much for this! I left two comments, one about some small typos and one about the embed viewing. I also saw some TODOs, but I think that it's no problem to merge these changes already if you're okay with it.
> 
> Thanks again, this is really helpful!

Thank you, happy I can be of help! If you're happy with the current way it's written and the fixes, please do merge it if you'd like! 😃

**Comment by CompuIves** [comment]
Great! This is already a huge improvement, thank you for this!

**Pull request merged**